File: go/publisher.go
package main

import (
    "fmt"
    "github.com/garyburd/redigo/redis"
)

type Publisher struct {
    channel string
    conn redis.Conn
}

func NewPublisher(channel string) Publisher {
    c, err := redis.Dial("tcp", ":6379")
    if err != nil {
        panic(fmt.Sprintf("PUBLISHER %s  -  Could not connect to Redis", channel))
    }

    return Publisher{
        conn: c,
        channel: channel,
    }
}

func (p Publisher) Publish(value string) {
    p.conn.Do("LPUSH", p.channel, value)
    p.conn.Do("PUBLISH", p.channel, 1)
}

File: erlang/publisher.erl
-module(publisher).
-export([publisher/1, publish/2]).

publisher(Channel) -> 
    {ok, Client} = eredis:start_link(),
    {Client, Channel}.

publish(Publisher, Value) ->
    {Client, Channel} = Publisher,
    [{ok, _}, {ok, _}] = eredis:qp(Client, [
        ["LPUSH", Channel, Value], 
        ["PUBLISH", Channel, 1]
    ]).
